### PR TITLE
Improved Detector Script and fixed some missing edge cases

### LINKED ISSUE #264: Add any_embedded_linux test to detect.py
I believe this will help reduce people forgetting it.

### PR DESCRIPTION
**Pull request by makermelissa**
Fixes #264.

### CODE REVIEW

File: adafruit_platformdetect/board.py [hunk 1]
```diff
@@ -752,6 +752,25 @@ def any_libre_computer_board(self) -> bool:
...
+            yield self.board.FTDI_FT2232H
+            yield self.board.MICROCHIP_MCP2221
+            yield self.board.BINHO_NOVA
+            yield self.board.GREATFET_ONE
+            yield self.board.PICO_U2IF
+            yield self.board.FEATHER_U2IF
+            yield self.board.ITSYBITY_U2IF
+            yield self.board.MACROPAD_U2IF
+            yield self.board.QTPY_U2IF
+            yield self.board.QT2040_TRINKEY_U2IF
```
Comment (tekktrik) [inline adafruit_platformdetect/board.py:770]: Is there any benefit to a generator here?

**Review by tekktrik — APPROVED** [review]
Looks good to me - one question on using a generator but not something that should gate approval.

File: adafruit_platformdetect/board.py (hunk 1 above)
Comment (makermelissa) [reply adafruit_platformdetect/board.py:770]: Mostly for readability, but also if it finds one, it stops checking.
